code/game/objects/items.dm

```diff
@@ -435,6 +447,29 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 			user.transfer_fingerprints_to(src)
 		else
 			to_chat(user, "<span class='notice'>You don't have enough tape to do that!</span>")
+	else if(istype(I, /obj/item/smithed_item/tool_bit) && can_have_bits)
+		if(length(attached_bits) >= max_bits)
+			to_chat(user, "<span class='notice'>Your tool already has the maximum number of bits!</span>")
+			return
+		var/obj/item/smithed_item/tool_bit/bit = I
+		bit.forceMove(src)
+		attached_bits += bit
+		bit.on_attached(user, src)
+
+/obj/item/AltClick(mob/user)
```

Comment:
Same here, this is really snowflakey. I think you might want to consider making this a component or something.